plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/navigation/openimplementation/OpenImplementationPresenter.java

```diff
@@ -96,161 +91,149 @@ public void show(final EditorPartPresenter editorPartPresenter) {
     final VirtualFile file = activeEditor.getEditorInput().getFile();
 
     if (file instanceof Resource) {
-      final Optional<Project> project = ((Resource) file).getRelatedProject();
+      final Project project = ((Resource) file).getProject();
 
       final Optional<Resource> srcFolder =
           ((Resource) file).getParentWithMarker(SourceFolderMarker.ID);
 
-      if (!srcFolder.isPresent()) {
+      if (project == null || !srcFolder.isPresent()) {
         return;
       }
 
       final String fqn = JavaUtil.resolveFQN((Container) srcFolder.get(), (Resource) file);
 
       service
-          .getImplementations(project.get().getLocation(), fqn, activeEditor.getCursorOffset())
+          .findImplementations(
```

Comment:
I think it would make more sense to pass the current location in the editor as a TextDocumentPositionParameter and to resolve the position to the innermost enclosing interface in the jdt.ls extension. That way, we can also find implementations of interfaces other than the main type. For example, we could select "Runnable" in "Runnable r= ...." and find implementers of that interface. Also, it would be more in line with general LSP conventions and we could easily extend it to find implementations of methods in the future.

Reply:
You can easily have two types called "Foo" in a Java project. Which one are you going to use?

Reply:
2 types can't have the same fully qualified name in one project